site/source/docs/porting/Debugging.rst

```diff
@@ -137,6 +137,45 @@ Debug printouts can even execute arbitrary JavaScript. For example::
   }
 
 
+Printing exception messages
+===========================
+
+Exceptions are thrown from WebAssembly using exception pointers, which means that try/catch/finally blocks in JavaScript will only receive a number, which represents a pointer into linear memory. In order to get the exception message, the user will need to create some WASM code which will extract the meaning from the exception, for example:
+
+.. code-block:: cpp
+
+  #include <bind.h>
+
+  std::string getExceptionMessage(int exceptionPtr) {
+    return std::string(reinterpret_cast<std::exception *>(exceptionPtr)->what());
+  }
```

Comment:
Could you do this without embind (which is fairly heavyweight construct).  

How about: 

```
EMSCRIPTEN_KEEPALIVE
const char* getExceptionMessage(int exceptionPtr) {
￼    return reinterpret_cast<const char*>(exceptionPtr)->what());
￼}
```

Then have JS call `UTF8ToString(getExceptionMessage(exception))`?

Then you don't need EMSCRIPTEN_BINDINGS either.

Reply:
In what sense is embind heavyweight? The doc seems to hint that it's reasonable, performance-wise.

Reply:
In what sense is embind heavyweight? The doc seems to hint that it's reasonable, performance-wise.

This question isn't relevant to the PR - I'd just like to know, since I use embind throughout my project.

Reply:
It makes sense to me to demonstrate one feature in isolation, when possible.   In order to understand this code one has to also know about how embind work.   It might lead the reader to think that embind is somehow essential to what is going on here. 

IMHO, just returning a `const char *` would make the code here simpler, with less boilderplate.  And there is no need to for the extra `--bind` link flag.

I think the reason I say heavyweight is because embind a whole bunch of C++ template magic that few people (myself included) have a good understanding of.  I don't know how costly it is a runtime (maybe its now), but it is pretty complex at compile time.  I'm not saying its bad or shouldn't be used,  I'd just rather not use when showing how other/simpler things work.

Reply:
I don't feel super strongly about not using embind here BTW.  This is just my opinion.    If you feel strongly that it helps here I'm not going block this PR.   I do appreciate contributions to the documentation.

Reply:
I tried recreating the code in your way, but couldn't get it to run.
I'm sorry - I assume I'm doing something wrong, but I can't really dig into it right now.

Reply:
Personally I think embind here is ok, if it makes things easier.

I can't seem to get this to work with embind, though. A full testcase might be helpful.